### ISSUE TITLE
Version information in morgue files surprising for released versions

### ISSUE THREAD

**Issue by Ozaq** [crawl/crawl#935]
Looking into a morgue file taken from underhound.eu from release 0.22  you will get the following version information:
`Dungeon Crawl Stone Soup version 0.22.1-2-g9c5b331383 (webtiles) character file.`

the expected output is:
`Dungeon Crawl Stone Soup version 0.22.1 (webtiles) character file.`

Reasoning:
gen_ver.pl does call
`git describe --tags`
which returns the last commit reachable by head and appends the distance in commits and the sha1
`<tag>-<distance>-g<sha1>`
This indicates that the builds running on the servers are not the commits tagged. The commits used to build the stable versions are actually the head commit of stone_soup-<MAJOR>.<MINOR>

Looking at the history of stone_soup-0.22 you can see:
```
9c5b331383 2018-09-18 15:08:00 -0400 advil (HEAD -> stone_soup-0.22, upstream/stone_soup-0.22) Revert complex bones install stuff, add dist_bones to repo
a44c0e56f4 2018-09-19 00:17:35 -0500 gammafunk Fix a message for wand pickup (minmay)
8f2143ff1b 2018-09-16 21:07:08 -0500 gammafunk (tag: 0.22.1) Update the debian changelog for 0.22.1
3f4de39d8c 2018-09-16 20:48:05 -0500 gammafunk Update the changelog for 0.22.1
402daae900 2018-09-11 20:07:57 -0400 CanOfWorms Update some scroll descriptions
```  

It looks like the underlying issue is just a mismatch between tag and head of the release branch. I am not sure how releases are coordinated with server owners but this should be very simple to fix by making sure the head of release branches is always appropriately tagged with a `<MAJOR>.<MINOR>.<PATCH>` tag.

Edit: Formatting ate unesacped `<MAJOR>.<MINOR>.<PATCH>`
Edit2: To be precise this is only surprising for stable versions, trunk and dev. versions are fine!

**Comment by alexjurkiewicz** [comment]
+1. Please keep released release branches HEAD pointing at the latest
release, it would simplify server admin

**Comment by Ozaq** [comment]
@alexjurkiewicz I think the head is pointing to the actual meant-to-be release commit. I seems that just the tag is off. But lets see what the others have to say :)

**Comment by rawlins** [comment]
I think that this behavior is correct; it results from pushing to the release branch after tagging a specific release version. Servers will update to the latest commit within a release branch as well. In particular, the `stone_soup-0.22` branch has two commits that have been pushed since 0.22.1 was tagged (one of them actually crucial to some server configs). In order to get them to line up, we'd need to release 0.22.2.

>Please keep released release branches HEAD pointing at the latest release, it would simplify server admin

Not sure I understand the request, how do we push commits to a release branch then?

**Comment by alexjurkiewicz** [comment]
By tagging the HEAD immediately afterwards :-)

**Comment by rawlins** [comment]
> By tagging the HEAD immediately afterwards :-)

I guess the idea is that a release is constructed by cherry-picking every commit for the release at once and tagging immediately, but that's not remotely what the process is. Even with a bulk cherry-pick, it is a ton of work that can take days, and isn't necessarily done by one person. Also, this idea would prevent pushing critical fixes to servers (like 9c5b33138319d) without tagging a release, which isn't viable.

Anyways, I'm going to close this.
